text/0000-extended-standard-library.md

```diff
@@ -0,0 +1,370 @@
+
+1. More confidence in the trustworthiness of the basic building blocks they rely on.
+2. An easier time finding the right building blocks.
+
+To achieve this, the Rust ecosystem needs a set of crates that includes the most commonly used functionality in Rust programs, referred to here as the ESL. It needs to be maintained by a trustworthy organization with a strong commitment to quality, security and support. As this set of crates matures, it should become possible to compose most Rust programs with a minimal number of dependencies outside of the ESL and the Rust standard library.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+The ESL will consist of a set of crates providing commonly used functionality.
+
+## Policies
+
+The ESL will require a number of policies, the most essential of which will ensure timely security fixes, strike a thoughtful balance between API stability and evolution, provide long-term support for older compilers through a careful MSRV choice while avoiding stagnation, and maintain a high standard for quality. The ESL does not need to make the same policy choices as the Rust Standard Library.
+
+A simplified list of example policies to use as potential starting point can be found in [Appendix A](#appendix-a). Policies may need to cover additional topics beyond the examples provided.
+
+Policies should evolve as the ESL community learns about what works best.
+
+## External Dependencies
+
+In an ideal world the ESL would have no production (non-dev) dependencies outside of the ESL and the Rust standard library. However, this is likely going to be impractical because an existing set of widely used crates with dependencies on each other already exists. If one or more of these crates does not want to join the ESL, exceptions may need to be made allowing them to be external dependencies in order to avoid expending a huge amount of effort to create and maintain duplicates of their functionality for the ESL. The ESL community should work to minimize the number of external dependencies for the sake of offering the most consistent, secure, and reliable experience possible, but some will almost certainly be necessary.
+
+## Selecting Functionality for Inclusion
+
+The primary criteria for selecting which functionality to include are:
+
+* Functionality that is widely used by the Rust ecosystem today, a pervasive dependency across many projects
+* Not duplicative of something that already exists in the Rust standard library or the ESL
+
+The following is an incomplete list of functionality that would likely be included based on these criteria, offered for the sake of roughly illustrating the intended scope of functionality:
+
+### Tranche 1 - Base functionality, leaf nodes in many dependency trees
+* ‘syn’ equivalent
+* ‘bitflags’ equivalent
+* ‘quote’ equivalent
+* Platform bindings (libc)
+* Encoding support (base64, hex, etc)
+* ‘cfg-if’ equivalent
```

Comment:
(nit) The use cases for `quote` and `cfg-if` are likely to be covered by `std` at some point in the hopefully-not-too-distant future: https://github.com/rust-lang/rust/issues/115585 https://github.com/rust-lang/rust/issues/54722